### PR TITLE
Adding Citrix plugin deprecation for M4 so it could be removed in M5

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Companion PR to #12333

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Try to configure CitrixPlugin, you should see a deprecation notice
3. Try to run SyncCommand, you should see a deprecation notice

### CODE REVIEW

**Comment by escopecz** [comment]
We did not understand each other. Apologies. I meant that you'd deprecate the Citrix plugin for Mautic 4.4.9 as Ruth suggested. We have to firstly deprecate the plugin in one version and then we can delete it in the next major version. We cannot remove it in 4.4.9 as it is a BC break.

This is the deprecation PR I did for Pipedrive for inspiration: https://github.com/mautic/mautic/pull/12364

**Comment by mabumusa1** [comment]
Sample outputs
image
image

File: phpstan-baseline.neon
Comment (mabumusa1) [inline phpstan-baseline.neon:92253]: @escopecz this error was ignored for other plugins so I ignored it as well

**Comment by mabumusa1** [comment]
> We did not understand each other. Apologies. I meant that you'd deprecate the Citrix plugin for Mautic 4.4.9 as Ruth suggested. We have to firstly deprecate the plugin in one version and then we can delete it in the next major version. We cannot remove it in 4.4.9 as it is a BC break.
> 
> This is the deprecation PR I did for Pipedrive for inspiration: #12364

No worries and thanks for pointing me to the right direction, I updated the PR as required

File: phpstan-baseline.neon
Comment (escopecz) [reply phpstan-baseline.neon:92253]: I'm OK with this since the goal of this PR is to remove the Citrix plugin altogether. Thanks for pointing this out!

**Review by escopecz — CHANGES_REQUESTED** [review]
@mabumusa1 please create also similar migration to https://github.com/mautic/mautic/pull/12364/files#diff-ac792d02497f71aedb711f6dd2540b2870677153a10b9bf74cd5a9cc196bffaaR12 that will add the deprecation messages to all users if the Citrix plugin is enabled in that Mautic instance.

@RCheesley @mollux Do we need an announcement blog post like we had for Pipedrive (https://github.com/mautic/mautic/pull/12364) or is it OK with deprecating it and suggesting an alternative plugin?

**Comment by RCheesley** [comment]
> @RCheesley @mollux Do we need an announcement blog post like we had for Pipedrive (#12364) or is it OK with deprecating it and suggesting an alternative plugin?

I think that if there is an alternative that is open source and freely available, it's helpful to raise more awareness of the process in a blog post which will reach a different audience. If that is the case maybe the maintainers of the plugin would be up for writing it?

**Comment by mabumusa1** [comment]
@escopecz I added the migration in this commit 095a3d2fa61ef5864f4776011f29a736b4c0c181

File: app/migrations/Version20230522115137.php
Comment (escopecz) [inline app/migrations/Version20230522115137.php]:
```suggestion
        $header  = 'The Citrix plugin will be removed in Mautic 5.0.0';
```

**Review by escopecz — COMMENTED** [review]
Once they forgotten plugin name is corrected this is good to go from code perspective. The command alert works:
```
bin/console mautic:citrix:sync

                                                                                                                        
 [WARNING] The Citrix plugin is deprecated and will be removed in Mautic 5. See <a                                      
           href="https://www.leuchtfeuer.com/en/mautic/know-how/mautic/gotowebinar-plugin-new-features/"                
           target=”_blank”>an alternative</a> for this plugin.
```
The notification is created.

And the warning appears in all 3 Citrix plugins:
Screenshot 2023-05-22 at 15 44 56

**Review by escopecz — DISMISSED** [review]
Good to go from code perspective. Thanks! 👍

**Comment by RCheesley** [comment]
Before this is merged can we just double check with Ekke and team that it's the right place to direct people (eg current URL or GitHub etc) before we push it out to everyone?

**Review by ekkeguembel — CHANGES_REQUESTED** [review]
Please change deprecation / replacement URL to https://www.leuchtfeuer.com/en/mautic/downloads/mautic-goto-plugin/
(Looks of that page preliminary, content is largely rendered from README.md which will be overhauled prior to 4.4.9... but URL is final : )

**Comment by mabumusa1** [comment]
> Please change deprecation / replacement URL to https://www.leuchtfeuer.com/en/mautic/downloads/mautic-goto-plugin/ (Looks of that page preliminary, content is largely rendered from README.md which will be overhauled prior to 4.4.9... but URL is final : )

Fixed in commit 5334fc3a1f911b3b4ba325a2ba0c8d9e7c2ab3c1

**Pull request merged**
